### PR TITLE
🧪 Blog App - Controllers Specs

### PR DESCRIPTION
**Pull request by DevKaranJ**
**In this project, your task is to set up tests for the controllers recently added to your Blog app.**

- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Ensure that for each action you check:

- [x] If the response status was correct.
- [x] If a correct template was rendered.
- [x] If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -3,14 +3,9 @@
...
-  end
-
-  describe 'GET /show' do
-    it 'returns http success' do
-      get '/posts/show'
+      # Assuming you have a user and some posts associated with that user
+      user = create(:user)
+      get user_posts_path(user)
       expect(response).to have_http_status(:success)
     end
```
Comment (thecodechaser) [inline spec/requests/posts_spec.rb]: - [x] Please make sure to check the below for each actions(index and show) for user and post controllers. Currently, you're checking only status. You should place the tests only in files re to posts and users, don't create other files for tests example `controllers_spec.rb`

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @DevKaranJ👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with controller specs. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -2,16 +2,12 @@
...
-      get '/posts/show'
-      expect(response).to have_http_status(:success)
+    it 'returns http success and renders the index template' do
+      user = create(:user)
+      get user_posts_path(user)
+      expect(response).to have_http_status(:success) # If response status was correct.
+      expect(response).to render_template(:index) # If the correct template was rendered.
+      expect(response.body).to include('List of Posts') # If the correct content was rendered.
     end
   end
```
Comment (thecodechaser) [inline spec/requests/posts_spec.rb]: - [x] Good job adding the missing tests for the post-index action as requested by me previously but you missed adding the tests for the post-show action. Please make sure to add all the above tests for post-show action as well and all tests should pass without any errors.

File: spec/requests/posts_spec.rb [hunk 3]
```diff
@@ -2,16 +2,12 @@
 
 RSpec.describe 'Posts', type: :request do
   describe 'GET /index' do
```
Comment (thecodechaser) [inline spec/requests/posts_spec.rb]: - [x] https://github.com/DevKaranJ/Blog/blob/1b443ff641535651becf7513e2f1c578ee7e5741/spec/requests/users_spec.rb#L3-L16
 
  - Please make sure to add other missing tests for the user index and user show actions as requested by me previously and all tests should pass without any errors.
 
image

File: spec/requests/posts_spec.rb [hunk 4]
```diff
@@ -2,16 +2,12 @@
...
-  describe 'GET /show' do
-    it 'returns http success' do
-      get '/posts/show'
-      expect(response).to have_http_status(:success)
+    it 'returns http success and renders the index template' do
+      user = create(:user)
+      get user_posts_path(user)
+      expect(response).to have_http_status(:success) # If response status was correct.
+      expect(response).to render_template(:index) # If the correct template was rendered.
+      expect(response.body).to include('List of Posts') # If the correct content was rendered.
```
Comment (thecodechaser) [inline spec/requests/posts_spec.rb]: - [x] All the tests are failing, Please make sure to fix it and all tests should pass without any errors.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @DevKaranJ👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- Good job with controller specs.
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @DevKaranJ, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Implemented the features as suggested by previous code reviewer ✅ 
- All tests are passing ✅
image


## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**